sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -116,6 +116,9 @@ trait DataFrame extends RDDApi[Row] {
   /** Prints the schema to the console in a nice tree format. */
   def printSchema(): Unit
 
+  /** Returns the plan information as string array for debugging purpose. */
+  def explain(extended: Boolean): Array[String]
```

Comment:
it seems to me it'd make more sense for explain to just print something to the console by default, isn't it?